.github/actions/free-hosted-runner-disk-space/action.yml

```diff
@@ -1,18 +1,156 @@
 name: "Free GHA runner disk space"
 description: "Removes unnecessary files from the runner."
 inputs:
+  only-remove-files:
+    description: 'Only remove files and folders, do not manipulate the runner memory'
+    required: false
+    default: 'true'
```

Comment:
IMO this needs a better name. Affirmative names rather than negative names are better and more intuitive. It is better to provide an optional flag to change the behaviour when its value is `true`, than providing an optional flag that you have to set to `false` to do the opposite of what the name says.

For example:
`merge-disk-volumes: true`

_(optional, default: false) to preserve backward compatibility_